data-prepper-plugins/saas-source-plugins/jira-source/src/test/java/org/opensearch/dataprepper/plugins/source/jira/JiraClientTest.java

```diff
@@ -0,0 +1,125 @@
+package org.opensearch.dataprepper.plugins.source.jira;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.source.source_crawler.base.CrawlerSourceConfig;
+import org.opensearch.dataprepper.plugins.source.source_crawler.base.PluginExecutorServiceProvider;
+import org.opensearch.dataprepper.plugins.source.source_crawler.coordination.state.SaasWorkerProgressState;
+
+import java.time.Instant;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class JiraClientTest {
+
+    @Mock
+    private Buffer<Record<Event>> buffer;
+
+    @Mock
+    private SaasWorkerProgressState saasWorkerProgressState;
+
+    @Mock
+    private CrawlerSourceConfig crawlerSourceConfig;
+
+    @Mock
+    private JiraSourceConfig jiraSourceConfig;
+
+    @Mock
+    ObjectMapper objectMapper;
+
+    @Mock
+    private JiraService jiraService;
+
+    @Mock
+    private JiraIterator jiraIterator;
+
+    private PluginExecutorServiceProvider executorServiceProvider = new PluginExecutorServiceProvider();
+
+    @Test
+    void testConstructor() {
+        JiraClient jiraClient;
+        jiraClient = new JiraClient(jiraService, jiraIterator, executorServiceProvider, jiraSourceConfig);
+        assertNotNull(jiraClient);
+    }
+
+    @Test
+    void testListItems() {
+        JiraClient jiraClient;
+        jiraClient = new JiraClient(jiraService, jiraIterator, executorServiceProvider, jiraSourceConfig);
+        assertNotNull(jiraClient.listItems());
+    }
+
+    @Test
+    void testSetLastPollTime() throws NoSuchFieldException, IllegalAccessException {
```

Comment:
Like we discussed, if this test is not really asserting anything then don't add it.

Reply:
ok